deliver/lib/deliver/upload_screenshots.rb

```diff
@@ -177,16 +178,22 @@ def wait_for_complete(iterator)
         is_processing = states.fetch('UPLOAD_COMPLETE', 0) > 0
         return states unless is_processing
 
+        if Time.now - start_time > timeout_seconds
+          UI.important("Consider incomplete screenshots failed after #{timeout_seconds} seconds")
```

Comment:
the wording of this message seems a bit off to me. Can we find a way to improve it?

Maybe `"Screenshots are still being processed after #{timeout_seconds} seconds. Aborting."` ?

Reply:
I agree that the message is weird here but I think the intent of what @mikhailmaslo meant is different than your suggestion @lacostej 👀 I think @mikhailmaslo meant is that the program will continue its execution (and not abort) but with "incomplete" screenshots, and that the user should consider them the same as "failed" 🤔 

Not sure if that's really the case as it's also confusing to me @mikhailmaslo could you clarify? If that's the case, can we improve the messaging here, and perhaps clarify what is going to happen? (Both for our understanding and documentation purposes, but also to set the expectation right to the user of this) 🙇

Reply:
Yeah, @rogerluan got my intention quite closely

The main idea here is to limit the waiting period for screenshots that are still in the processing state after the specified timeout is reached. Upon reaching this timeout, these screenshots are treated as if they have failed to process correctly. This means they will be automatically deleted and then re-uploaded

A better message message could be

```Suggestion
          UI.important("Timeout of #{timeout_seconds} seconds for screenshots processing has been reached. Aborting.")
```

What do you think @lacostej @rogerluan?
If it looks good to you, I will make this fix and re-request your reviews

Reply:
"Aborting" sounds like the program will stop with an error 😬  perhaps we could write something close to what you just described to us? e.g.:

```suggestion
          UI.important("Screenshot upload reached the timeout limit of #{timeout_seconds} seconds. We'll now retry uploading the screenshots that couldn't be uploaded in time.")
```